### PR TITLE
Start of tree cleanup

### PR DESCRIPTION
**Pull request by TheColorRed**
This is the start of cleaning up the treeBuilder to remove duplicate code.

### CODE REVIEW

**Comment by TheColorRed** [comment]
This also fixes "Comma Separated Constants"

File: server/src/hvy/treeBuilder.ts [hunk 1]
```diff
@@ -900,39 +798,36 @@ export class TreeBuilder
         {
             var funcNode: FunctionCallNode = new FunctionCallNode();
 
-            if (codeLevel[1][0] == "ns")
-            {
+            if (codeLevel[1][0] == "ns" && Array.isArray(codeLevel[1]) && Array.isArray(codeLevel[1][1])) {
```
Comment (nevadascout) [inline server/src/hvy/treeBuilder.ts]: The `Array.isArray(codeLevel[1])` check should be before `codeLevel[1][0] == "ns"`

File: server/src/suggestionBuilder.ts [hunk 2]
```diff
@@ -6,12 +6,16 @@
...
-import { TextDocumentPositionParams, TextDocument, CompletionItem, CompletionItemKind } from 'vscode-languageserver';
+import { TextDocumentPositionParams, TextDocument, CompletionItem, CompletionItemKind, IConnection, createConnection,
+    IPCMessageReader, IPCMessageWriter
+} from 'vscode-languageserver';
 import {
     TreeBuilder, FileNode, SymbolType, AccessModifierNode,
     ClassNode, TraitNode, MethodNode
 } from "./hvy/treeBuilder";
 
+let connection: IConnection = createConnection(new IPCMessageReader(process), new IPCMessageWriter(process));
```
Comment (nevadascout) [inline server/src/suggestionBuilder.ts]: Is this variable actually being used for anything?

Comment (TheColorRed) [inline server/src/suggestionBuilder.ts]: I was using it for debugging some things, I think I removed where I was doing the debugging.

File: server/src/hvy/treeBuilder.ts (hunk 1 above)
Comment (TheColorRed) [inline server/src/hvy/treeBuilder.ts]: Alright, I will fix.

**Pull request merged**